globocom/gitlab-ci-monitor
# [Feature] Adding tests support

# Feature
- [x] Added `Jest` testing library (and other support ones);
- [x] Configured the environment to support testing features;
- [x] Added sample tests to guarantee critical functionalities (needs future improvements);
- [x] Added mock files to simplify tests;

# Steps to run the tests on this branch
* Checkout the branch
* `npm install`
* `npm test`

This solves #4.

---

[Jest Docs](https://jestjs.io/docs/en/getting-started)

⚠️ It is important to disclaim that increasing further testability, it is necessary to refactor the application to reduce method _side effects_. That should secure more stability to every step of the assertion, although, due to the size of this change, it might be a good idea to submit a separate PR to attack that.
